lib.ml

```diff
@@ -683,6 +683,11 @@ let env : (string * (located_primitive * Types.datatype * pure)) list = [
    datatype "(Xml) ~> mu n.[ [|Text:String | Attr:(String, String) | Node:(String, n) |] ]",
    IMPURE);
 
+  "variantToXml",
+  (`Server (p1 (fun v -> `List (List.map (fun i -> `XML (i)) (Value.xml_of_variants v)))),
+   datatype "(mu n.[ [|Text:String | Attr:(String, String) | Node:(String, n) |] ]) ~> Xml",
```

Comment:
```
typename XmlItem = [| Text: String | Node: (String, Map(String, String), [ XmlItem ]) |]
```
would be a better alternative, but since that will most likely require changes in a few more place, I'd say it is out of scope for this PR.